ggml-metal.m

```diff
@@ -1732,20 +1729,20 @@ static enum ggml_status ggml_metal_graph_compute(
                         //       indirect matrix multiplication
                         // !!!
                         if ([ctx->device supportsFamily:MTLGPUFamilyApple7] &&
-                            ne20 % 32 == 0 && ne20 >= 64 &&
+                            ne00 % 32 == 0 && ne00 >= 64 &&
                             ne11 > ne11_mm_min) {
 
                             // some Metal matrix data types require aligned pointers
                             // ref: https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf (Table 2.5)
-                            switch (src0->type) {
-                                case GGML_TYPE_F32: GGML_ASSERT(nb01 % 16 == 0); break;
-                                case GGML_TYPE_F16: GGML_ASSERT(nb01 % 8  == 0); break;
+                            switch (src2->type) {
+                                case GGML_TYPE_F32: GGML_ASSERT(nb21 % 16 == 0); break;
+                                case GGML_TYPE_F16: GGML_ASSERT(nb21 % 8  == 0); break;
```

Comment:
On `master` this is incorrect - I made a mistake in: https://github.com/ggerganov/llama.cpp/pull/6225

On this branch it should become:

```suggestion
                            switch (src0->type) {
                                case GGML_TYPE_F32: GGML_ASSERT(nb01 % 16 == 0); break;
                                case GGML_TYPE_F16: GGML_ASSERT(nb01 % 8  == 0); break;
```